subsys/bluetooth/host/conn.c

```diff
@@ -1895,6 +1925,42 @@ int bt_conn_get_info(const struct bt_conn *conn, struct bt_conn_info *info)
 	return -EINVAL;
 }
 
+#if defined(CONFIG_BT_REMOTE_VERSION)
+static int hci_le_read_remote_version(struct bt_conn *conn)
+{
+	int err;
+	struct bt_hci_cp_read_remote_version_info *cp;
+	struct net_buf *buf;
+
+	if (atomic_test_and_set_bit(conn->flags,
+				    BT_CONN_PENDING_VERSION_INFO)) {
+		return -EBUSY;
+	}
+	
+	buf = bt_hci_cmd_create(BT_HCI_OP_READ_REMOTE_VERSION_INFO,
+				sizeof(*cp));
+	if (!buf) {
+		atomic_clear_bit(conn->flags, BT_CONN_PENDING_VERSION_INFO);
+		k_sem_give(&conn->rv.sem);
```

Comment:
Why k_sem_give here? 
I don't see why it is needed here. Maybe I'm missing something @jhedberg?
Same for the one below.

Reply:
If there is an error or no buffer, we won't actually get to the process of retrieving the remote version and as such it won't actually be pending and we clear the flag.

Reply:
I understand the flag, I'm asking about the k_sem_give. We haven't acquired it.

Reply:
Ah, then to answer you question "why are they there?" I made a mistake that will be corrected now.